### PR TITLE
fix subscribe panic 

### PR DESCRIPTION
**Pull request by nogang**
## Proposed changes

- If there are a lot of subscribed connections, intermittently the node dies due to panic. This is the PR that corrects this.
- The cause was a problem caused by delivering an event that occurred on a connection in a disconnected state.
## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: networks/rpc/subscription.go [hunk 1]
```diff
@@ -133,3 +136,11 @@ func (n *Notifier) activate(id ID, namespace string) {
 		delete(n.inactive, id)
 	}
 }
+
+// unsubscribe unsubscribe every subscriptions.
+func (n *Notifier) unsubscribe() {
+	n.subMu.Lock()
+	defer n.subMu.Unlock()
+
+	n.active = make(map[ID]*Subscription)
```
Comment (KimKyungup) [inline networks/rpc/subscription.go]: We need to `close(s.err)` for all subscriptions before removing subscriptions.

File: networks/rpc/subscription.go [hunk 2]
```diff
@@ -107,9 +107,9 @@ func (n *Notifier) Closed() <-chan interface{} {
 	return n.codec.Closed()
 }
 
-// unsubscribe a subscription.
+// unsubscribeWithID a subscription.
```
Comment (jeonghonah) [inline networks/rpc/subscription.go]: Minor comments:
Method description is easy to understand.

File: networks/rpc/subscription.go [hunk 3]
```diff
@@ -133,3 +133,14 @@ func (n *Notifier) activate(id ID, namespace string) {
 		delete(n.inactive, id)
 	}
 }
+
+// unsubscribe unsubscribe every subscriptions.
+func (n *Notifier) unsubscribe() {
```
Comment (KimKyungup) [inline networks/rpc/subscription.go]: How about `unsubscribeAll`? :)

File: networks/rpc/subscription.go [hunk 4]
```diff
@@ -133,3 +133,14 @@ func (n *Notifier) activate(id ID, namespace string) {
 		delete(n.inactive, id)
 	}
 }
+
+// unsubscribeAll unsubscribe every subscriptions.
```
Comment (KimKyungup) [inline networks/rpc/subscription.go]:
```suggestion
// unsubscribeAll unsubscribes every subscriptions.
```

**Comment by jeonghonah** [comment]
It will be nice if we add a unit test in future PR.

**Pull request merged**